### DISCUSSION TITLE
Updating DXVK/Vulkan/MoltenVK?

### DISCUSSION THREAD

**Discussion by asmaloney** [The-Wineskin-Project/WineskinServer#239] [Q&A] [answered]
My use-case is Guild Wars 2 which now uses DX11. I'm using CX21.1 and I ran _dxvk_macos.verb_. The game works alright (not awesome, but mostly playable). I know there's a new version of the [Vulkan SDK](https://vulkan.lunarg.com/) released since then (v1.3.204.1).

How does DXVK (which I assume uses the Vulkan SDK through MoltenVK?) get updated? Is it something I do or is it only updated when WineCX21.x is released?

Will we always need to run that .verb or will it eventually just be included as part of the WineCX release?

(More generally I have a bad mental model of how all the wine/wineskin/winetricks/Vulkan stuff fits together, so if this doesn't make sense, that's why :-))

Thank you for all of this!

**Comment by Gcenx** [comment, marked as answer]
> My use-case is Guild Wars 2 which now uses DX11. I'm using CX21.1 and I ran _dxvk_macos.verb_. The game works alright (not awesome, but mostly playable). 

I’ve forked winetricks and it now listed macOS compatible versions of DXVK.

> I know there's a new version of the [Vulkan SDK](https://vulkan.lunarg.com/) released since then (v1.3.204.1).

The Vulkan SDK can’t be used as a custom patched version of MoltenVK is required to fake some required extensions to make DXVK-macOS happy, this will be getting bumped to the latest build in the new wrapper update.

This can be manually updated by grabbing the latest version from my MoltenVK mirror https://github.com/Gcenx/MoltenVK/releases and replacing the copy found within the wrapper /Frameworks directory.

> How does DXVK (which I assume uses the Vulkan SDK through MoltenVK?) get updated? Is it something I do or is it only updated when WineCX21.x is released?

Until recently I was making use of CodeWeavers provided DXVK-1.7 however I’m now providing newer versions, thought I don’t foresee there being too much difference as most too the issues are related Metal lacking needed features.

> Will we always need to run that .verb or will it eventually just be included as part of the WineCX release?

See my above answer the custom winetricks fork will now display compatible DXVK-macOS package once I’ve compiled/uploaded them.

**Reply by asmaloney** [reply]
Thank you so much!

**Comment by asmaloney** [comment]
I tried downloading [macos_dxvk_patched-1.1.8.tar.xz](https://github.com/Gcenx/MoltenVK/releases/download/v1.1.8/macos_dxvk_patched-1.1.8.tar.xz) from [here](https://github.com/Gcenx/MoltenVK/releases/tag/v1.1.8) and dropping in the new _libMoltenVK.dylib_.

This gave me an error when trying to run it:

<img width="712" alt="Screen Shot 2022-04-02 at 13 41 48 PM" src="https://user-images.githubusercontent.com/391371/161396250-dcbae2cc-06f4-435c-8783-d9cf5cb63a68.png">

I switched back to the old one and it works agin.

Maybe some kind of ABI incompatibility? Was there a step I'm missing?

(I just wanted to follow up. I don't want to take away your dev time on this - I can just wait for the next release!)

**Reply by Gcenx** [reply]
By chance do you have an Intel GPU?

**Comment by asmaloney** [comment]
Nope - I'm running an iMac with a Radeon Pro 580X 8 GB.

**Edit:**
macOS 10.14.6
3.6 GHz Intel Core i9
32 GB RAM

**Reply by Gcenx** [reply]
Very strange as I had removed an export that should have only affected Intel GPUs and not AMD GPUs, I’ll comment something you could try once I’m home
